modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java

```diff
@@ -488,7 +490,10 @@ private static void addLocalSubscriptionsForAllBindingsOfQueue(AMQQueue queue, S
                         Andes.getInstance().openLocalSubscription(subscriptionEvent);
                         alreadyAddedSubscriptions.add(subscriptionEvent);
                     }
+                } else {
+                    throw new AndesException("Binding already exists for the queue " + queue.getName());
```

Comment:
Related to your previous reply.

> This is QpidBridge class. Also we throw this exception due to an limitation in andes code. This is not a violation  of amqp spec. So I feel like throwing an Andes exception is better here. WDYT?

I don't think this is a limitation in Andes core, and it's better to have a clear separation between Andes core and transports. Everything before QpidBridge should be unaware of classes in Andes core and everything after QpidBridge should be unaware of transport specific classes. Only the bridge should be aware of classes in both sides. Therefore -1 in using AndesException.